src/gennodejs/generate.py

```diff
@@ -618,7 +618,7 @@ def get_dynamic_field_length_line(field, query):
                                 raise Exception('Unexpected field {} with type {} has unknown length'.format(f.name, f.base_type))
                             # it's a string array!
                             len_constant_length_fields += 4
-                            line_to_write = 'length += object.{}.length;'.format(f.name)
+                            line_to_write = 'length += Buffer.byteLength(object.{}, \'utf8\');'.format(f.name)
```

Comment:
Yikes! There should be a `_getByteLength` function in scope that I meant to use for this with the minor benefit of being able to update it without re-releasing `gennodejs`, which is kind of a pain. If you're able to make that update and verify it still works for you that'd be great.

Reply:
i don't see the connection between `ros_msg_utils` and this repo. is the intent that when consuming the types that `gennodejs` generates, you should also take the `ros_msg_utils` repo and attach those exported declarations to the global object? just trying to get a sense of how these things are intended to be used together.

Reply:
Yep - the code generated by [`gennodejs`](https://github.com/RethinkRobotics-opensource/gennodejs/blob/kinetic-devel/src/gennodejs/generate.py#L283) (and even by [`rosnodejs`](https://github.com/RethinkRobotics-opensource/rosnodejs/blob/a151f0369b4f12efabc9538daf19f9fbe5e3af83/src/utils/messageGeneration/MessageWriter.js#L93)) assumes that `ros_msg_utils` is available in the global namespace. `rosnodejs` [puts it there](https://github.com/RethinkRobotics-opensource/rosnodejs/blob/a151f0369b4f12efabc9538daf19f9fbe5e3af83/src/utils/message_utils.js#L27) and since the output of `gennodejs` is generally intended just to be used with `rosnodejs` it should "just work".

I hoped to eventually find a better way around that issue but it was never a priority.